sdks/python/apache_beam/transforms/util.py

```diff
@@ -864,3 +867,225 @@ def add_window_info(element, timestamp=DoFn.TimestampParam,
 
     def expand(self, pcoll):
       return pcoll | ParDo(self.add_window_info)
+
+
+class Regex(object):
+  """
+  PTransform  to use Regular Expression to process the elements in a
+  PCollection.
+  """
+  @staticmethod
+  def _regex_compile(regex):
+    """Return re.compile if the regex has a string value"""
+    if isinstance(regex, str):
+      regex = re.compile(regex)
+    return regex
+
+  @staticmethod
+  @ptransform_fn
+  def _matches_all_object(pcoll, regex, group=None):
+    """
+    PTransform to find the matches as per the regex.
+    """
+    def _process(element):
+      m = regex.finditer(element)
+      results = list()
+      for _, match in enumerate(m, start=1):
```

Comment:
Why the enumerate?

Reply:
just used my old snippet which also uses the index, in our case indexes are not required, I will update this as well.